extension/checkout/lockUserCheckout.js

```diff
@@ -0,0 +1,49 @@
+const uuidv4 = require('uuid/v4')
+const InternalError = require('./../common/Error/InternalError')
+const CheckoutInProgressError = require('./../common/Error/CheckoutInProgressError')
+
+/**
+ * Locks down a checkout to a max of once every 20 seconds on a single device or user
+ *
+ * @param {SDKContext} context
+ * @param {Object} input
+ * @param {number} input.lockDuration
+ * @throws {CheckoutInProgressError|InternalError}
+ * @returns {Promise<ExtCheckoutLock>}
+ */
+module.exports = async (context, input) => {
+  const storageName = 'user'
+  const dataStorageKey = 'checkoutLock'
+  const lockDuration = input.lockDuration * 1000 // convert seconds to milliseconds
+
+  const lockId = uuidv4()
+  const timestamp = Math.floor(Date.now() / 1000) * 1000
```

Comment:
We can store milliseconds from `(new Date()).getTime()` and then just compare with  `(new Date())`

Reply:
That's already happening, It's just floored down to full seconds.